ext/MitsubaLoader/CMitsubaLoader.cpp

```diff
@@ -1288,27 +1329,22 @@ void instr_execute_PLASTIC(in instr_t instr, in uvec3 regs, in mat2 dUV, in bsdf
 	vec3 refl = textureOrRGB19E7(data.data[1].yz, instr_getPlasticReflTexPresence(instr), dUV)*scale.z;
 	registers[REG_DST(regs)] = refl;
 }
-void instr_execute_ROUGHPLASTIC(in instr_t instr, in uvec3 regs, in mat2 dUV, in bsdf_data_t data)
+void instr_execute_ROUGHPLASTIC(in instr_t instr, in uvec3 regs, in mat2 dUV, in vec2 a, in float D, in bsdf_data_t data)
 {
 	if (currBSDFParams.isotropic.NdotL>FLT_MIN)
 	{
-		uint ndf = instr_getNDF(instr);
 		vec3 eta = vec3(uintBitsToFloat(data.data[0].x));
 		vec3 eta2 = eta*eta;
 		vec3 scale = decodeRGB19E7(uvec2(data.data[1].w, data.data[2].x));
 		vec3 refl = textureOrRGB19E7(data.data[1].yz, instr_getPlasticReflTexPresence(instr), dUV)*scale.z;
-		float a = textureOrF32(data.data[0].yz, instr_getAlphaUTexPresence(instr), dUV)*scale.x;
-		float a2 = a*a;
 
+		float a2 = a.x*a.y;
```

Comment:
maybe move the `a2` to the CPU side (run an image filter to square the roughness values)

Reply:
would this be possible in the new architecture?

Reply:
ofc, nothing really changes here, but it's just one multiplication. + often `a` (not squared) is needed so that would generate a need to do a sqrt

Reply:
ah ok, the sqrt.. fine then